### DISCUSSION TITLE
Pbench Agent / Server API compatibility story for 0.69 to 0.71 to 0.72 transition

### DISCUSSION THREAD

**Discussion by portante** [distributed-system-analysis/pbench#2216] [Ideas]
* Next out the door is 0.71 agent that talks to a 0.69 server
 * Next is 0.71 server talks 0.69/0.71 agent SSH protocol
 * Next is 0.72 agent & server which only talk to each other via RESTful APIs
   * But carries either a shim or uses the satellite server infrastructure to allow for 0.71 and earlier agents to send data to 0.72 servers
 * There would be a time window the admin can set for retirement of that shim/satellite setup
 * Future versions of the server can choose to support pervious versions of the _new_ RESTful API as appropriate

**Comment by dbutenhof** [comment]
Issue #2100 discusses prep-shim transition
Issue #2097 discusses satellite server integration issues
